hw_4.js

```diff
@@ -0,0 +1,329 @@
+    isHappy: true,
+}
+
+function getKeyObject(myObj) {
+    for (let key in myObj) {
+        console.log(key);
+    }
+}
+
+function getValueObject(myObj) {
+    for (let key in myObj) {
+        console.log(myObj[key]);
+    }
+}
+
+getKeyObject(student);
+getValueObject(student);
+
+//Task 10
+
+const colors = {
+    'ru pum pu ru rum': {
+        red: 'красный',
+        green: 'зелёный',
+        blue: 'синий',
+    }
+}
+
+console.log(colors['ru pum pu ru rum'].red);
+console.log(colors['ru pum pu ru rum'].blue);
+
+//Task11 
+
+let salaries = {
+    andrey: 500,
+    sveta: 413,
+    anton: 987,
+    lena: 664,
+    alexandra: 199,
+}
+
+function getAvgSalary(myObj) {
+    let i = 0;
+    let sum = 0;
+    for (let key in myObj) {
+        i++;
+        sum += myObj[key];
+    }
+    return sum / i;
+}
+let avgsalary = getAvgSalary(salaries);
+console.log(avgsalary);
+
+
+//Task 12
+
+function user(login, password) {
+    this.login = login.trim();
+    this.password = password.trim();
+}
+
+let myUser = new user(prompt('Введите свой логин'), prompt('Введите свой пароль'));
+
+function isRigth(myLogin, myPassword, currUser) {
+    myLogin.trim() === currUser.login && myPassword.trim() === currUser.password ? console.log('Добро пожалоВать') : console.log('Такого логина и пароля не существует!');
+}
+
+isRigth(prompt('Подтвердите свой логин'), prompt('Подтвердите свой пароль'), myUser);
+
+//Task 13
+
+const myColors = ['red', 'green', 'blue'];
+
+console.log(myColors.length);
+
+//Task 14
+const animals = ['mankey', 'dog', 'cat'];
+
+console.log(animals.at(-1));
+
+//Task 15
+
+const numbers = [5, 43, 63, 23, 90];
+
+console.log(numbers);
+
+// способ №1
+function deleteArray(myArray) {
+    for (let i = 0; i < myArray.length; i++) {
+        delete myArray[i];
+    }
+}
+
+// способ №2
+numbers.length = 0;
+
+console.log(numbers);
+
+//Task 16
+
+const students = ['Polina', 'Dasha', 'Masha'];
+
+students.pop();
+students.push('Borya');
+students.shift();
+students.unshift('Andrey');
+
+console.log(students);
+
+//Task 17
+
+const cats = ['Gachito', 'Tom', 'Batman'];
+
+function getArrayFor(myArray) {
+    for (let i = 0; i < myArray.length; i++) {
+        console.log(myArray[i]);
+    }
+}
+
+function getArrayForOf(myArray) {
+    for (let i of myArray) {
+        console.log(i);
+    }
+}
+
+getArrayFor(cats);
+getArrayForOf(cats);
+
+//Task 18
+const evenNumbers = [2, 4, 6, 8, 10];
+const oddNumbers = [1, 3, 5, 7, 9];
+
+const newNumbers = evenNumbers.concat(oddNumbers);
+
+console.log(newNumbers);
+console.log(newNumbers[8]);
+
+//Task 19
+const binary = [0, 0, 0, 0];
+
+function getString(myArray) {
+    let str = '';
+    for (let i = 0; i < myArray.length; i++) {
+        i === myArray.length - 1 ? str += myArray[i] : str += myArray[i] + '1';
+    }
+    return str;
+}
+
+console.log(getString(binary));
+
+//ADVANCED level
+//Task 1
+
+function isTriangle(a, b, c) {
+    return a + b > c && a + c > b && c + b > a ? true : false;
```

Comment:
тут тернарник лишний
так как результат сравнения и так будет типа boolean
```suggestion
    return a + b > c && a + c > b && c + b > a;
```